### PR TITLE
EOS-24947: log dir needs to be updated with pod_id(machine_id)

### PR DESCRIPTION
**Pull request by suryakumar1024**
# Problem Statement
Log files should be created inside a separate directory for all the pods.

In the k8s environment, pod_id will be machine_id. We need to add a directory named, {machine_id} inside every component log location

Created a generic method inside common module that should return log dir path till machine_id-> /var/log/cortx/utils/{machine_id}

After path update log are created as expected below

[root@ssc-vm-g2-rhev4-1158 32d2605f5b8d4aec8f16ca084b6f1a86]# pwd
/tmp/log/test/cortx/utils/32d2605f5b8d4aec8f16ca084b6f1a86

[root@ssc-vm-g2-rhev4-1158 32d2605f5b8d4aec8f16ca084b6f1a86]# ls
iem  message_bus  utils_server  utils_setup.log

# Design
-  For Bug describe the fix here. 
-  For Feature, Post the link to the solution page on the confluence CORTX Foundation Library 

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [x] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/common/common.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import os
+from cortx.utils import const
+from cortx.utils.conf_store import Conf
+
+
+class CortxConf:
+
+    def load_config() -> None:
```
Comment (selvakumaar5496) [inline py-utils/src/utils/common/common.py:8]: make this function internal

File: py-utils/src/utils/common/common.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import os
+from cortx.utils import const
+from cortx.utils.conf_store import Conf
+
+
+class CortxConf:
```
Comment (selvakumaar5496) [inline py-utils/src/utils/common/common.py:6]: Store index in a variable

File: py-utils/src/utils/common/common.py (hunk 1 above)
Comment (suryakumar1024) [reply py-utils/src/utils/common/common.py]: updated in recent patch

File: py-utils/src/utils/common/common.py (hunk 2 above)
Comment (suryakumar1024) [reply py-utils/src/utils/common/common.py:6]: Updated in recent patch

File: py-utils/src/utils/common/common.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @staticmethod
+    def _load_config() -> None:
+        """Load cortx.conf file into conf in-memory"""
+        Conf.load(CortxConf._index, f'json://{const.CORTX_CONF_FILE}',
+            skip_reload=True)
+
+    @staticmethod
+    def get_log_path(component = None, base_dir: str = None) -> str:
+        """
+        Get the log path with machine-id/ pod-id as sub directory
```
Comment (ujjwalpl) [inline py-utils/src/utils/common/common.py:18]: Please avoid using pod or any k8s construct in the code.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/common/common.py (hunk 3 above)
Comment (suryakumar1024) [reply py-utils/src/utils/common/common.py]: Updated comment in the recent patch

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/common/common.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    @staticmethod
+    def set_key(key: str, value: str):
+        """Sets the value into conf in-memory at the given key"""
+        CortxConf._load_config()
+        return Conf.set(CortxConf._index, key, value)
+
+    @staticmethod
+    def save():
+        """Saves the configuration into the cortx.conf file"""
+        return Conf.save(CortxConf._index)
```
Comment (selvakumaar5496) [inline py-utils/src/utils/common/common.py:47]: Does save return anything ?

File: py-utils/src/utils/common/common.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                   Default = None
+        base_dir: root directory where all the log sub-directories should be create.
+        """
+        CortxConf._load_config()
+        log_dir = base_dir if base_dir else Conf.get(CortxConf._index, 'log_dir')
+        return os.path.join(log_dir, f'cortx/utils/{Conf.machine_id}'\
+            +f'{"/"+component if component else ""}')
+
+    @staticmethod
+    def get_key(key: str, default_val: str = None, **filters):
```
Comment (selvakumaar5496) [inline py-utils/src/utils/common/common.py:33]: I think this interface should be renamed as get_value

File: py-utils/src/utils/common/common.py (hunk 4 above)
Comment (suryakumar1024) [reply py-utils/src/utils/common/common.py]: Updated in recent patch

File: py-utils/src/utils/common/common.py (hunk 5 above)
Comment (suryakumar1024) [reply py-utils/src/utils/common/common.py:33]: @selvakumaar5496 but it given a purposeful meaning if we call it as get_key/ set_key as we are setting/ getting a key into a conf store.

Comment (suryakumar1024) [reply py-utils/src/utils/common/common.py]: As discussed i updated the method name to get and set

**Review by selvakumaar5496 — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**